libobs/obs-internal.h

```diff
@@ -1089,6 +1090,66 @@ struct caption_track_data {
 	struct deque caption_data;
 };
 
+struct counter_data {
+	uint32_t diff;
+	uint32_t ref;
+	uint32_t curr;
+};
+
+// Broadcast Performance Metrics frame timing
+struct bpm_frame_time {
```

Comment:
The idea of not calling this `bpm_frame_time` seems like a good idea. When I implemented it, it was part of the overall BPM work and hadn't realized it might be useful for more generic uses, so this makes sense. `encoder_packet_timing` is a good name, however some of the timing info is related to the uncompressed frames, so it's not totally accurate. The frame times straddle the uncompressed frames and compressed packets, so it's a bit tricky to name. I'll think about this a bit more, and we may end up using `encoder_packet_timing`.